casper/src/main/scala/coop/rchain/casper/api/GraphGenerator.scala

```diff
@@ -80,40 +83,27 @@ object GraphzGenerator {
                 )
             )
         // create invisible edges from ancestors to first node in each cluster for proper alligment
-        _ <- validators.toList.traverse {
+        _ <- validatorsList.traverse {
               case (id, blocks) =>
                 allAncestors.traverse(ancestor => {
                   val node = nodeForTs(id, firstTs, blocks, lastFinalizedBlockHash)._2
                   g.edge(ancestor, node, style = Some(Invis))
                 })
             }
         // draw clusters per validator
-        _ <- validators.toList.traverse {
+        _ <- validatorsList.traverse {
               case (id, blocks) =>
                 g.subgraph(
                   validatorCluster(id, blocks, timeseries, lastFinalizedBlockHash)
                 )
             }
+        // draw parent dependencies
+        _ <- drawParentDependencies[G](g, validatorsList)
         // draw justification dotted lines
-        _ <- validators.values.toList.flatMap(_.values.toList).traverse {
-              case ValidatorBlock(blockHash, parentsHashes, justifications) => {
-                val parentsEdges = parentsHashes
-                  .traverse(p => g.edge(blockHash, p, constraint = Some(false)))
-                val justificationsEdges = justifications
-                  .traverse(
-                    j =>
-                      g.edge(
-                        blockHash,
-                        j,
-                        style = Some(Dotted),
-                        constraint = Some(false),
-                        arrowHead = Some(NoneArrow)
-                      )
-                  )
-                parentsEdges *> justificationsEdges
-              }
-
-            }
+        _ <- config.showJustificationLines.fold(
+              drawJustificationDottedLines[G](g, validators),
+              ().pure[G]
```

Comment:
Is this line necessary?

Reply:
yes
by default we will *NOT* draw justifictactions, only if `showJustificatioLines` flag is `true`
so here i'm folding over that flag, if `true` i draw, if `false` `(().pure[G])` which is a NOP